### ISSUE TITLE
cql_clustering_key_desc together with CLUSTERING ORDER BY (field DESC)

### ISSUE THREAD

**Issue by svok** [strapdata/elassandra#264]
**Elassandra version**: 6.2.3.10

**Plugins installed**: [] (no)

**JVM version** (`java -version`):  "version": "1.8.0_144", "vm_name": "Java HotSpot(TM) 64-Bit Server VM",

**OS version** (`uname -a` if on a Unix-like system): "name": "Linux",  "arch": "amd64",  "version": "4.14.83-gentoo"

**Description of the problem including expected versus actual behavior**:

We are moving our keyspaces from Elassandra 5.5.0.8 to 6.2.3.10. For that we create a new datacenter DC3 and share the keyspace from the old datacenter DC1 to DC3. Now we are applying a modified version of the Elasticsearch index settings and mappings and this causes the problem.

Our table contains DESC type ordering and it cannot be properly handled by the new setting `cql_clustering_key_desc`. All other settings work properly. As I change ordering from DESC to ASC, the index is created with no errors. But DESC option causes the error message:
```json
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "Failed to execute query:null : Existing column [change_date_t] type [text] mismatch with inferred type [text]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to execute query:null : Existing column [change_date_t] type [text] mismatch with inferred type [text]",
    "caused_by": {
      "type": "configuration_exception",
      "reason": "Existing column [change_date_t] type [text] mismatch with inferred type [text]"
    }
  },
  "status": 400
}
```
Changing the `cql_clustering_key_desc` setting to both true and false or removing this option at all gives no effect.

**Steps to reproduce**:

1. Apply cassandra cql schema
2. Apply Elasticsearch json schema - error

* elassandra logs (logs/system.logs or /var/lib/cassandra/system.log)
elasticsearch log
```
2019-02-08 13:31:46,106 INFO  [MigrationStage:1] ColumnFamilyStore.java:410 <init> Initializing test_ks.test_t
2019-02-08 13:34:18,982 WARN  [elasticsearch[195.19.11.218][masterService#updateTask][T#1]] MasterService.java:684 executeTasks failed to execute cluster state update source [create-index [test_ks], cause [api]]                                                                                                                                               org.elasticsearch.index.mapper.MapperParsingException: Failed to execute query:null : Existing column [change_date_t] type [text] mismatch with inferred type [text]
        at org.elassandra.cluster.SchemaManager.updateTableSchema(SchemaManager.java:751)
        at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$IndexCreationTask.execute(MetaDataCreateIndexService.java:571)
        at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:52)
        at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:676)
        at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:300)
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:221)
        at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:152)
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
        at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:573)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.cassandra.exceptions.ConfigurationException: Existing column [change_date_t] type [text] mismatch with inferred type [text]
        at org.elassandra.cluster.ColumnDescriptor.validate(ColumnDescriptor.java:81)
        at org.elassandra.cluster.SchemaManager.updateTableSchema(SchemaManager.java:739)
        ... 14 common frames omitted
```

* elasticsearch cluster state (curl http://localhost:9200/_cluster/state)
I will provide it if required

* cassandra schema (cqlsh>DESC KEYSPACE <your_keyspace>)
```cql
create keyspace if not exists test_ks with durable_writes = true
	and replication = {'class': 'org.apache.cassandra.locator.NetworkTopologyStrategy', 'DC3': '3'};

CREATE TABLE IF NOT EXISTS test_ks.test_t (
    ep_number text,
    change_date_t text,

   PRIMARY KEY (ep_number, change_date_t)
 )
WITH CLUSTERING ORDER BY (change_date_t DESC);
```
* elasticsearch schema
```json
{
    "mappings": {
      "test_t": {
        "_source": {
          "enabled": true
        },
        "properties": {
           "ep_number": {
	     "type": "keyword",
             "cql_collection": "singleton",
             "cql_partition_key" : true,
             "cql_primary_key_order": 0
          },
          "change_date_t": {
             "type": "keyword",
             "cql_collection": "singleton",
	     "cql_clustering_key_desc": true,
             "cql_primary_key_order": 1
          }
        }
      }
    },
    "settings": {
      "index": {
        "number_of_replicas": "1"
      }
    }
  }
```

* cassandra gossip state (run: nodetool gossipinfo)
I will provide it if required

**Comment by vroyer** [comment]
Hi.
This is a bug, fixed in the next release 6.2.3.11 planned for tomorrow.
Thanks.

**Comment by vroyer** [comment]
This is fixed in release 6.2.3.11 now available.
Thanks.

**Comment by svok** [comment]
I confirm, now it works well. Thank you.
